interop-tests/src/bin/wasm_ping.rs

```diff
@@ -0,0 +1,210 @@
+use std::env;
+use std::process::Stdio;
+use std::time::Duration;
+
+use anyhow::{anyhow, Context, Result};
+use axum::http::header;
+use axum::response::IntoResponse;
+use axum::routing::get;
+use axum::{extract::State, http::StatusCode, routing::post, Json, Router};
+use redis::{AsyncCommands, Client};
+use thirtyfour::prelude::*;
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tower_http::cors::CorsLayer;
+use tower_http::services::ServeDir;
+use tower_http::trace::TraceLayer;
+use tracing::{error, warn};
+use tracing_subscriber::{fmt, prelude::*, EnvFilter};
+
+use interop_tests::BlpopRequest;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // start logging
+    tracing_subscriber::registry()
+        .with(fmt::layer())
+        .with(EnvFilter::from_default_env())
+        .init();
+
+    // read env variables
+    let redis_addr = env::var("redis_addr")
+        .map(|addr| format!("redis://{addr}"))
+        .unwrap_or_else(|_| "redis://redis:6379".into());
+    let proxy_addr = env::var("proxy_addr").unwrap_or_else(|_| "127.0.0.1:6378".into());
+    let proxy_addr_clone = proxy_addr.clone();
+    let wasm_pkg_dir = env::var("wasm_pkg_dir").unwrap_or_else(|_| "interop-tests/pkg".into());
+    let test_timeout = Duration::from_secs(
+        env::var("test_timeout_seconds")
+            .unwrap_or_else(|_| "180".into())
+            .parse::<u64>()?,
+    );
```

Comment:
This will not be as straightforward. We technically can have a single binary, but we'll need to switch it's logic by an addition of some feature or maybe an env var. The reason for that is that only the library is built either for wasm or for host but the binaries are both built for the host target. In case of host, the binary just calls the main function from the lib. In case of wasm, we compile the library to the wasm and then on host we need to setup HTTP server to serve our wasm package, redis proxy to allow it to speak to redis, webdriver to load our hosted wasm to the browser.
I don't see any sane way to make this selection automatic, so if you prefer a single binary approach then please let me know whether you like some feature or the env var more?

Reply:
Ah I see. I guess specifying _which_ binary to build is already the config value we are looking for.

In that case, can we at least strip as much duplication from the binaries as possible? Like the env variable parsing for example.

We can make a module for that that is only activated by the binaries and not the library part of the package.

Reply:
done